### PR TITLE
Ocaml 5 on runtime4: stdlib (except 46aea8c924)

### PR DESCRIPTION
**Pull request by mshinwell**
This should be everything from the stdlib backports, except the stdlib part of `46aea8c924`, which conflicts badly and I will look at separately.

### CODE REVIEW

File: stdlib/bigarray.mli [hunk 1]
```diff
@@ -120,6 +120,7 @@ type ('a, 'b) kind =
   | Complex32 : (Complex.t, complex32_elt) kind
   | Complex64 : (Complex.t, complex64_elt) kind
   | Char : (char, int8_unsigned_elt) kind (**)
+  | Float16 : (float, float16_elt) kind
```
[the remark this answers is not in the document]
Comment (xclerc) [reply stdlib/bigarray.mli:123]: (If we keep the line, the `(**)` from the
previous line should be moved to that
one.)

File: configure.ac [hunk 2]
```diff
@@ -653,9 +654,8 @@ AS_CASE([$host],
     [exeext=".exe"],
   [exeext=''])
 
-otherlibraries="dynlink runtime_events"
-lib_dynlink=true
```
[the remark this answers is not in the document]
Comment (mshinwell) [reply configure.ac:657]: It's unclear this variable is used in the tree, but I've resurrected this line.

File: configure.ac [hunk 3]
```diff
@@ -653,9 +654,9 @@ AS_CASE([$host],
     [exeext=".exe"],
   [exeext=''])
 
-otherlibraries="dynlink runtime_events"
+dnl otherlibraries="dynlink runtime_events"
+otherlibraries="dynlink"
 lib_dynlink=true
-lib_runtime_events=true
```
Comment (xclerc) [inline configure.ac:658]: Sorry, I should have posted that earlier:
I wonder whether it would be safer to
simply `dnl` this line?

**Review by xclerc — APPROVED** [review]

File: configure.ac (hunk 3 above)
Comment (mshinwell) [reply configure.ac:658]: Sure, I've left CRs at this point too now.

Comment (xclerc) [reply configure.ac:658]: Indeed, I should have suggested
that, without the CRs, the `dnl`
provides little safety.

**Pull request merged**